main.js

```diff
@@ -19,58 +20,78 @@ function run() {
             // Get inputs
             const dt_upload_api_key = core.getInput("DT_UPLOAD_API_KEY");
             const input_binary_path = core.getInput("UPLOAD_BINARY_PATH");
+            // Mask the API key
             core.setSecret(dt_upload_api_key);
-            if (!fs.existsSync(input_binary_path)) {
-                throw new Error("Input file does not exist at " + input_binary_path);
+            // Check that the inputs are set
+            if (!dt_upload_api_key) {
+                throw new Error("DT_UPLOAD_API_KEY must be set!");
             }
-            // retry upload 3 times
-            for (let loop_idx = 0; loop_idx < 3; loop_idx++) {
-                // Send the auth request to get the upload URL
-                const auth_response = yield fetch("https://api.securetheorem.com/uploadapi/v1/upload_init", {
-                    method: 'POST',
-                    headers: {
-                        Authorization: "APIKey " + dt_upload_api_key,
-                        Accept: "application/json",
-                        "Content-Type": "application/json",
-                    },
-                });
-                let auth_json;
-                try {
-                    auth_json = yield auth_response.json();
-                }
-                catch (err) {
-                    core.setFailed(err);
-                }
-                if (auth_response.status !== 200) {
-                    // handles auth failure
-                    core.setFailed(auth_json);
-                    break;
-                }
-                const form = new FormData();
-                form.append('file', fs.createReadStream(input_binary_path));
-                // Send the scan request with file
-                const response = yield fetch(auth_json.upload_url, {
-                    method: 'POST',
-                    body: form,
-                });
-                let jsonformat;
-                try {
-                    jsonformat = yield response.json();
-                }
-                catch (err) {
-                    core.setFailed(err);
-                }
-                // Check the response
-                console.log(jsonformat);
-                if (response.status === 200) {
-                    core.setOutput('response', jsonformat);
-                    ;
-                    break;
+            if (!input_binary_path) {
+                throw new Error("UPLOAD_BINARY_PATH must be set!");
+            }
+            const files = glob.sync(input_binary_path);
+            if (!files.length) {
+                throw new Error("Did not find any files that match path:" + input_binary_path);
+            }
+            console.log("Found files matching path: " + input_binary_path);
+            console.log(files);
+            // Upload all the files that matched the file path
```

Comment:
I think this could be a problem (uploading multiple files), but not sure?

Reply:
Maybe we should add a limit of 2 or 3?

Reply:
Uploading only 2 or 3 out of (many) is not ideal because there is no way to control which of the (many) files get uploaded. 

I think we should either error out if more than one file is matched, or upload 'em all. I vote to leave it as is. That way it is possible to build multiple apps off the same code base and upload them all with one call to the plugin.